examples/demo-app/src/analytics.js

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// This file sends actions on the demo app to Google analytics
+
+import {ActionTypes} from 'kepler.gl/actions';
+import {LOCATION_CHANGE} from 'react-router-redux';
+
+const getPayload = ({payload}) => payload;
+
+const withStoreInformation = getTracking => (payload, store) => {
+  const trackingFromPayload = getTracking(payload, store);
+  try {
+    const {
+      demo: {
+        keplerGl: {
+          map: {
+            visState: {datasets = {}, filters = [], layers = []}
+          }
+        }
+      }
+    } = store.getState();
+    const trackingFromStore = {
+      datasetsCount: Object.keys(datasets).length,
+      layersCount: layers.length,
+      filtersCount: filters.length
+    };
+    return {
+      ...trackingFromStore,
+      ...trackingFromPayload
+    };
+  } catch (err) {
+    /* eslint-disable */
+    console.warn(err);
+    /* eslint-enable */
+    return trackingFromPayload;
+  }
+};
+
+const trackingInformation = {
+  [ActionTypes.LOAD_FILES]: ({files}) =>
+    files.map(({size, type}) => ({size, type})),
+  [ActionTypes.LAYER_HOVER]: ({
+    payload: {
+      info: {lngLat}
+    }
+  }) => lngLat,
+  [ActionTypes.LAYER_CONFIG_CHANGE]: ({
+    payload: {
+      oldLayer: {type},
+      newConfig
+    }
+  }) => ({
+    type,
+    newConfig
+  }),
+  [ActionTypes.MAP_STYLE_CHANGE]: getPayload,
+  [ActionTypes.TOGGLE_MODAL]: getPayload,
+  [ActionTypes.TOGGLE_SIDE_PANEL]: getPayload,
+  [ActionTypes.UPDATE_MAP]: withStoreInformation(getPayload),
+  [ActionTypes.SET_FILTER]: withStoreInformation(getPayload),
```

Comment:
assume this is where we track the type of filter being used. payload would be not wrapped. and it would sed {idx, prop, value}
```
const {prop} = payload;
if (prop === 'name') {
// get the type of the filter from the next store
const visState = store.demo.keplerGl.map.visState;
const filter  = visState.filters[idx]
const newType = filter.type;

const trackingPayload = {type: newType}
}

```